### ISSUE TITLE
Demo support: explore possible options

### ISSUE THREAD

**Issue by michaeldgg2** [OpenE2140/OpenE2140#339]
Demo versions do not contain all the necessary content that OpenE2140 needs (mainly assets related to new units/buildings in MP2, but also the new tileset), so it definitely can't be a drop-in replacement to the assets from the full game.

The goal here is to explore possible options on how to support demo version, so that players, who would want to try OpenE2140 before buying full version of Earth 2140, could be able to do so.

One option could be making placeholders for all actors, sequences, assets etc. that require data from assets, which need full version. Then limiting actors, maps and other features that would be available, when installing assets from demo version. This means that technically MP games with demo assets would have to be separate from MP games with full assets.

This might not be possible (without changes to OpenRA engine) so another solution involving two (demo and full) OpenRA mods, that share most of the stuff, might be necessary.

**Comment by Dzierzan** [comment]
IMO that's a `wontfix` scenario. Imagine the amount of work needed to cover demo support. Earth 2140 is quite cheap and if it goes on sale, it becomes dirt cheap.

OpenE2140 can be easily treated as a mod and what do you need to use a mod? That's right, the full game.

**Comment by michaeldgg2** [comment]
You're right about the price, when it's on sale, it's really cheap. But there might be small group of people, who can't (or don't want to) buy games on Steam/GOG. Also another (small) advantage is that if we supported demo version(s), we could make it easily downloadable from inside the mod and thus making it more accessible. I personally like it that default OpenRA mods are playable with just little effort (they can download freeware versions of RA/CNC/TS).

My opinion is that we could leave it open, maybe someone can pick this up in the future.

**Comment by IceReaper** [comment]
Well, if all tilesets are compatible between eachother, the demo will at least have one. So some tileset loading magic may simply redirect any tileset to the one provided. As the speciality on E2140 tilesets is that they are swappable.

Also when detecting the demo, we could disable loading the campaign, but instead allow to play the demo mission only in singleplayer.

Other thing is missing art for sprites. It would be possible to add in a "stupid" empty box sprite which is used instead with its unit type being an icon in the middle.

Especially if you would add a lobby option to "disable addon content". People could then play with the demo and explicitely disable missing addon content to avoid seeing such rectangle-placeholder-units.

If you list all the stuff which would be missing, a solution could be worked out here. I would say its definitely possible. And if someone gets hookes, he is pretty much encouraged to buy the full version for the full experience.

**Comment by Dzierzan** [comment]
Sounds like more work than needed. Earth 2140 is not expensive now and with a sale, it becomes super cheap.
